android-components/components/browser/icons/src/main/java/mozilla/components/browser/icons/decoder/SvgIconDecoder.kt

```diff
@@ -6,55 +6,67 @@ package mozilla.components.browser.icons.decoder
 
 import android.content.Context
 import android.graphics.Bitmap
-import androidx.core.graphics.drawable.toBitmap
-import coil.ImageLoader
-import coil.decode.SvgDecoder
-import coil.executeBlocking
-import coil.request.CachePolicy
-import coil.request.ImageRequest
+import android.graphics.Bitmap.Config.ARGB_8888
+import android.graphics.Canvas
+import android.graphics.RectF
+import androidx.core.graphics.createBitmap
+import com.caverock.androidsvg.SVG
+import com.caverock.androidsvg.SVGParseException
 import mozilla.components.support.base.log.logger.Logger
 import mozilla.components.support.images.DesiredSize
 import mozilla.components.support.images.decoder.ImageDecoder
 
 /**
- * [ImageDecoder] that will use the Coil library [SvgDecoder.Factory] in order to decode the byte data.
+ * [ImageDecoder] that will use the AndroidSVG in order to decode the byte data.
  *
- * ⚠️ For guidance on use of the Coil library see comment for [ComponentsDependencies.thirdparty_coil_svg].
  */
 class SvgIconDecoder(val context: Context) : ImageDecoder {
     private val logger = Logger("SvgIconDecoder")
 
     override fun decode(data: ByteArray, desiredSize: DesiredSize): Bitmap? =
         try {
-            val request = ImageRequest.Builder(context)
-                .size(desiredSize.targetSize)
-                .data(data)
-                .build()
+            val svg = SVG.getFromInputStream(data.inputStream())
 
-            SvgImageLoader.getInstance(context).executeBlocking(request).drawable?.toBitmap()
+            val svgWidth: Float
+            val svgHeight: Float
+            val viewBox: RectF? = svg.documentViewBox
+            if (viewBox != null) {
+                svgWidth = viewBox.width()
+                svgHeight = viewBox.height()
+            } else {
+                svgWidth = svg.documentWidth
+                svgHeight = svg.documentHeight
+            }
+
+            var bitmapWidth = desiredSize.targetSize
+            var bitmapHeight = desiredSize.targetSize
+
+            if (svgWidth > 0 && svgHeight > 0) {
+                val widthPercent = bitmapWidth / svgWidth.toDouble()
```

Comment:
Nice 👌 
Perhaps a comment on this block will be useful for future maintainers? Maybe something like "Scale the bitmap to SVG maintaining the aspect ratio"